nvbench/detail/entropy_criterion.cxx

```diff
@@ -88,15 +100,53 @@ void entropy_criterion::do_add_measurement(nvbench::float64_t measurement)
 
     if (it != m_freq_tracker.end() && it->first == key)
     {
+      old_count = it->second;
       it->second += 1;
+      new_count = it->second;
     }
     else
     {
+      old_count = 0;
       m_freq_tracker.insert(it, std::make_pair(key, nvbench::int64_t{1}));
+      new_count = 1;
     }
   }
 
-  m_entropy_tracker.push_back(compute_entropy());
+  update_entropy_sum(old_count, new_count);
```

Comment:
Note that `new_count` is always `old_count + 1`, so perhaps there is no need to introduce `new_count` variable.